secure-php-auth/includes/functions.php

```
@@ -0,0 +1,579 @@
+    regenerateSession();
+    return true;
+}
+
+function isLoggedIn() {
+    return isset($_SESSION['user_id']) && checkSessionTimeout();
+}
+
+/**
+ * Rate limiting functions
+ */
+function checkRateLimit($key, $maxAttempts, $timeWindow) {
```

PHPMD (2.15.0)

Avoid unused parameters such as '$timeWindow'. (Unused Code Rules)

(UnusedFormalParameter)